src/main/java/baritone/process/BuilderProcess.java

```diff
@@ -150,10 +151,13 @@ public void buildOpenSchematic() {
             Optional<Tuple<IStaticSchematic, BlockPos>> schematic = SchematicaHelper.getOpenSchematic();
             if (schematic.isPresent()) {
                 IStaticSchematic s = schematic.get().getFirst();
+                BlockPos origin = schematic.get().getSecond();
+                ISchematic schem = Baritone.settings().mapArtMode.value ? new MapArtSchematic(s) : s;
+                schem = Baritone.settings().schematicaOnlyBuildSelection.value ? new SelectionSchematic(schem, origin, baritone.getSelectionManager().getSelections()) : schem;
```

Comment:
Maybe move this to a separate function, this would mean that the same code could be used in both buildOpenSchematic and buildSchematic.

Reply:
Yea, thought about this and then decided that refactoring other people's code is not what this pr is there for.